tests/modules/text_generation/test_text_generation.py

```diff
@@ -70,7 +79,46 @@ def test_save_model_can_run():
     with tempfile.TemporaryDirectory() as model_dir:
         model.save(model_dir)
         del model
-        new_model = TextGeneration.load(model_dir, load_backend=StubBackend())
-        sample_text = "Hello stub"
+        new_model = TextGeneration.load(model_dir, load_backend=StubTGISBackend())
+        result = new_model.run(SAMPLE_TEXT, preserve_input_text=True)
+        assert isinstance(result, GeneratedTextResult)
+        assert result.generated_text == "moose"
+        assert result.generated_tokens == 1
+        assert result.finish_reason == 5
+
+
+### Output streaming tests ##############################################################
+
+
+def test_bootstrap_and_run_stream_out():
+    """Check if we can bootstrap and run_stream_out"""
+    model = TextGeneration.bootstrap(SEQ2SEQ_LM_MODEL, load_backend=StubTGISBackend())
+
+    stream_result = model.run_stream_out(SAMPLE_TEXT)
+    assert isinstance(stream_result, Iterable)
+    # Convert to list to more easily check outputs
+    result_list = list(stream_result)
+    assert len(result_list) == 3
+    first_result = result_list[0]
+    assert first_result.generated_text == "moose"
+    assert first_result.tokens[0].text == "moose"
+    assert first_result.tokens[0].logprob == 0.2
+    assert first_result.details.finish_reason == 5
+    assert first_result.details.generated_tokens == 1
+    assert first_result.details.seed == 10
+
+
+def test_run_stream_out_with_runtime_error():
+    """Check if runtime error from client raises"""
+
+    with mock.patch.object(StubTGISClient, "GenerateStream") as mock_gen_stream:
+        mock_gen_stream.side_effect = RuntimeError("An error!")
+
+        model = TextGeneration.bootstrap(
+            SEQ2SEQ_LM_MODEL, load_backend=StubTGISBackend()
+        )
         with pytest.raises(RuntimeError):
-            new_model.run(sample_text, preserve_input_text=True)
+            response = model.run_stream_out(SAMPLE_TEXT, preserve_input_text=True)
+            for _ in response:
```

Comment:
This should crash immediately, right? Can you change this to `next(response)` instead?